plugins/UsersManager/API.php

```diff
@@ -107,7 +107,16 @@ public function getUserPreference($userLogin, $preferenceName)
             return $optionValue;
         }
 
-        return $this->getDefaultUserPreference($preferenceName, $userLogin);
+        $defaultPreference = $this->getDefaultUserPreference($preferenceName, $userLogin);
+
+        // A `get*` method should not set anything. But in this case we do it to improve performance as it prevents
+        // us from loading all siteIds (which can be 50k or more) the next time this preference is requested.
+        // this method can be called quite often when generating links etc (to get defaultWebsiteId).
+        if ($defaultPreference !== false && $preferenceName === self::PREFERENCE_DEFAULT_REPORT) {
+            $this->setUserPreference($userLogin, $preferenceName, $defaultPreference);
+        }
```

Comment:
Would that be an issue for the time before the scheduled task is run? Ie, for 24 hours there would be poor performance. Or poor-er performance.

I will also default the user preference when creating/updating users.